R/app_server.R

```diff
@@ -125,15 +125,114 @@ app_server <- function(input, output, session) {
 
   observeEvent(data_r$click, {
     callModule(mod_DT_server, "DT_ui_participants",
-      data = data_r$participants %>%
-        dplyr::select(-dplyr::any_of(getOption("emdash.cols_to_remove_from_participts_table"))) %>%
-        data.table::setnames(originalColumnNames, new_column_names, skip_absent = TRUE)
+               data = data_r$participants %>%
+                 dplyr::select(-dplyr::any_of(getOption("emdash.cols_to_remove_from_participts_table"))) %>%
+                 data.table::setnames(originalColumnNames, new_column_names, skip_absent = TRUE)
     )
     callModule(mod_DT_server, "DT_ui_trips",
-      data = data_r$trips %>%
-        dplyr::select(-dplyr::any_of(getOption("emdash.cols_to_remove_trips_table"))) %>%
-        sf::st_drop_geometry()
+               data = data_r$trips %>%
+                 dplyr::select(-dplyr::any_of(getOption("emdash.cols_to_remove_trips_table"))) %>%
+                 sf::st_drop_geometry()
     )
+    
+    tableList <- getOption('emdash.supplementary_tables')
+    
+    # For each supplementary table, append a new tabPanel and run the server function
+    # that specifies table behavior
+    for (t in tableList){
+      table_type <- names(t)
+      table_title <- t[[table_type]]$tab_name
+      
+      # If the table is Bike Check In, use dtedit
+      if (table_type == 'Checkinout') {
+        
+        # Define the callback functions used by dtedit
+        #' Insert a row. "Create"
+        #' @param data the data including your inserted row
+        #' @param row the row where you made the change
+        insert_callback <- function(data, row) {
+          db_insert(cons,"Checkinout",data[row,])
+          return(data)
+        }
+        
+        #' Update a row
+        #' @param data the data including your updated row
+        update_callback <- function(data, olddata, row) {
+          db_update(cons,"Checkinout",data[row,])
+          data
+        }
+        
+        #' Delete a row
+        delete_callback <- function(data, row) {
+          db_delete(cons,"Checkinout",data[row,])
+          return(data[-row, ])
+        }
+        
```

Comment:
I will deploy this for now, but I was hoping that this would also be configurable. So as part of the config file, we would have a flag which said that the supplementary table was "editable". If it was, we would wrap it in an editable `DT_edit`. Note that we already have the table name in the config, can't we just use the table name here instead of hardcoding `Checkinout`?

Reply:
This should be made into a module, but for some reason we couldn't make it work with DTedit (if I remember this correctly).

Reply:
@asiripanich there is a more recently updated fork of DTEdit that claims that it is modularized (https://github.com/DavidPatShuiFong/DTedit). Maybe @allenmichael099 can look into that?
The same user submit a PR to the main DTEdit, but it hasn't been merged yet (https://github.com/jbryer/DTedit/pull/5)